docs/source/new-project.rst

```diff
@@ -176,7 +176,7 @@ Of course, nothing is stopping you from using forward from within the training_s
 
     def training_step(self, batch, batch_idx):
         ...
-        z = self(x)
+        z = self.forward(x)
```

Comment:
I think we can call `self`, @awaelchli?

Reply:
`self(x)` works too. It's a pytorch thing. Not a bug.

Reply:
agree, this is pytorch and going through call is preferred over forward. If you think the docs are not clear, then maybe we can consider changing the sentence above slightly, like so:

> Of course, nothing is stopping you from using forward (e.g. through ``__call__``) from within the training_step. 

But I would hold back on explaining too many pytorch specific things in our docs.